miden/tests/integration/operations/u32_ops/arithmetic_ops.rs

```diff
@@ -34,16 +34,19 @@ fn u32add_fail() {
 
     // should fail if a >= 2^32
     let test = build_op_test!(asm_op, &[U32_BOUND, 0]);
-    test.expect_error(TestError::ExecutionError("FailedAssertion"));
+    test.expect_error(TestError::ExecutionError("NotU32Value"));
 
     // should fail if b >= 2^32
     let test = build_op_test!(asm_op, &[0, U32_BOUND]);
-    test.expect_error(TestError::ExecutionError("FailedAssertion"));
+    test.expect_error(TestError::ExecutionError("NotU32Value"));
 
     // should fail if a + b >= 2^32
     let a = u32::MAX;
     let b = 1_u64;
     let test = build_op_test!(asm_op, &[a as u64, b]);
+    // TODO: Discuss if the error thrown in handle_arithmetic_operation
+    // should be changed to NotU32Value somehow instead of FailedAssertion
+    // which is thrown when the result >= 2^32
```

Comment:
Great point! Unfortunately, I don't see an easy way to do this without either significant impacting performance or refactoring the way we execute operations. For now, I'd remove this comment and create a separate issue for this.

Reply:
Created #218 for tracking this, thanks